Add --no-daemonize option to synctl

SohamG opened the pull request:
Fix #4130 

- Add parser argument "--no-daemonize"

Signed-off-by: sohamg <sohamg2@gmail.com>

### Pull Request Checklist

<!-- Please read CONTRIBUTING.rst before submitting your pull request -->

* [ ✅] Pull request is based on the develop branch
* [ ✅] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#changelog)
* [✅ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

erikjohnston commented:
This looks good, thanks! The only issue I can see is that this will not do the right thing if the daemonize option has been set in the config file. I think this is probably fine for now, though it would be great if you could either a) note in the argument description it only works if `daemonize` is not set in the config, or b) change the `--daemonize` config option in synapse to support both `--daemonize` and `--daemonize=no` so that we can then use that form for the `--no-daemonize` case

erikjohnston commented on synctl:
```diff
@@ -69,10 +69,14 @@ def abort(message, colour=RED, stream=sys.stderr):
 
 
-def start(configfile):
+def start(configfile, daemonize = True):
```
PEP8 is failing due to:

```
[2019-06-10T12:39:25Z] synctl:72:32: E251 unexpected spaces around keyword / parameter equals
```

SohamG replied:
Fixed in 0afcbc6

SohamG commented:
Hey, thanks for the review. I fixed the pep8 errors, and implemented option a) in your message for the time being. I'd like to implement option b too, it might take me some time though. Should option b be for another PR?

erikjohnston commented:
Your last commit looks to have reintroduced the pep8 error. Once that's fixed I'm happy to merge this PR if you don't think you'll have time in the next couple of days to do option b.
